### PR TITLE
Musical content

### PR DESCRIPTION
**Pull request by PropaneSoup**
This utilizes the new musical foundations we have and adds some new instruments, and a new music store building for them all to spawn in. The new instruments are an acoustic guitar, flute, clarinet, saxophone, trumpet, tuba, ukulele, banjo, violin and a super-rare 'golden fiddle' easter egg, à la the Charlie Daniels Band.

I also went in and fixed some lack of tabbing at the bottom of tools.json to match the rest of the file.

### CODE REVIEW

File: data/json/item_groups.json [hunk 1]
```diff
@@ -823,6 +823,13 @@
       ["purse", 40],
       ["mbag", 20],
       ["fanny", 10],
+	  ["acoustic_guitar", 20],
```
Comment (KA101) [inline data/json/item_groups.json]: I was worried when I saw the "lack of tabbing" comment.  Borne out here.  Notepad++, probably other reputable editors such as vim or emacs, can set the Tab key to insert spaces.  Tabs as such display differently on different editors and lead to frustrated maintainers.

File: data/json/items/tool_armor.json [hunk 2]
```diff
@@ -2163,5 +2163,94 @@
...
+            "descriptions": [
+                "You strum a few chords on your guitar.",
+                "You pluck a few notes on your guitar.",
+                "You play a slow blues-y tune on your guitar."
+            ]
+        },
+        "revert_to": "null",
+        "storage" : 0,
+        "symbol" : "-",
+        "description" : "A simple wooden acoustic guitar with a strap so you can wear it while you aren't playing",
```
Comment (KA101) [inline data/json/items/tool_armor.json]: Surely whilst you're playing it, too?

File: data/json/items/tools.json [hunk 3]
```diff
@@ -9482,360 +9482,656 @@
...
+        "type": "musical_instrument",
+        "speed_penalty": 13,
+        "volume": 14,
+        "fun": -5,
+        "fun_bonus": 2,
+        "description_frequency": 100,
+        "descriptions": [
+            "You play a little tune on your clarinet.",
+            "You play a moving, orchestral piece on your clarinet.",
+            "You let out piercing shriek from your clarinet. Oops."
```
Comment (KA101) [inline data/json/items/tools.json]: a piercing shriek, and that should really make a different noise.

File: data/json/item_groups.json (hunk 1 above)
Comment (PropaneSoup) [inline data/json/item_groups.json]: Hmm, I've heard about that, and that's why I *thought* I only used spaces for these.

File: data/json/items/tool_armor.json (hunk 2 above)
Comment (PropaneSoup) [inline data/json/items/tool_armor.json]: Oh yea :P Little misleading in the description. I'll make it a little more accurate.

File: data/json/items/tools.json (hunk 3 above)
Comment (PropaneSoup) [inline data/json/items/tools.json]: Ah, fair enough. As much as I'd like to keep it, that's beyond my abilities, at least for now. I'll change the message.

File: data/json/regional_map_settings.json [hunk 4]
```diff
@@ -87,7 +87,8 @@
               "s_jewelry":2,
               "s_antique": 2,
               "s_gardening": 2,
-              "museum": 1
+              "museum": 1,
+			  "s_music": 5
```
Comment (KA101) [inline data/json/regional_map_settings.json]: More tab problems.

File: data/json/mapgen/musicstore.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                "c": "f_chair",
+                "S": "f_sink",
+                "d": "f_dumpster",
+                "z": "f_null"
+            },
+            "toilets": {
+                "T": { }
+            },
+            "add": [
+                { "item": "violin_golden", "x": 21, "y": 8, "chance": 95}
```
Comment (KA101) [inline data/json/mapgen/musicstore.json]: 95 seems pretty high?

Comment (PropaneSoup) [inline data/json/mapgen/musicstore.json]: From my playtesting, it seemed like the higher the number, the less likely a spawn. I believe that means that there's only a 5% chance to spawn.

**Comment by KA101** [comment]
Now my name's KA101,
and I've already committed unforgiveable sin,
your PR's something we won't regret
but it's not *yet* the best there's ever been.

File: data/json/mapgen/musicstore.json (hunk 5 above)
Comment (BevapDin) [inline data/json/mapgen/musicstore.json]: It's a `one_in( chance )` check, 95 means pretty rare. This is apparently the only place in mapgen that uses this kind of logic.

Comment (PropaneSoup) [inline data/json/mapgen/musicstore.json]: Ah, so using my quick math, that's a 1% chance to spawn. Maybe a little too rare then?

Comment (KA101) [inline data/json/mapgen/musicstore.json]: Nah, they shouldn't be common.  Easter-egg critters are one in 10,000 shots, by comparison.

File: data/json/items/tools.json [hunk 6]
```diff
@@ -9482,360 +9482,656 @@
...
+    "volume" : 6,
+    "cutting" : 0,
+    "phase" : "solid",
+    "bashing" : 4
+  },
+  {
+    "type" : "TOOL",
+    "id" : "saxophone",
+    "name" : "saxophone",
+    "name_plural": "saxophones",
```
Comment (KA101) [inline data/json/items/tools.json]: I played sax in elementary-school band.  Worn with neck strap, so gonna suggest tool_armor treatment.

**Comment by PropaneSoup** [comment]
When KA101 finished PropaneSoup said
"Well you're pretty good at reviewing son,
But click that merge button over there
Cuz this PR's as good as done!"

**Comment by KA101** [comment]
The instruments should probably give messages every few turns: if I'm making noise, I should know about it more often than (roughly) once every five minutes.  I'll tweak the sax's strap wording and fix the tabs in the merge.

**Comment by PropaneSoup** [comment]
> The instruments should probably give messages every few turns

This can be done already, just lower the description_frequency entry. Putting it to 1 seemed to make it give a message every turn. Twenty seems like a good value as I'm testing it now, if you wanna change those in the merge as well

**Pull request merged**